chapters/functions.tex

```diff
@@ -2645,6 +2645,27 @@ \subsection{Utility Functions}\label{utility-functions}
 
 This section describes the utility functions declared in \filename{ModelicaUtilities.h}, which can be called in external Modelica functions written in C.
 
+The tool must ensure that the header is found by \lstinline[language=C]!#include "ModelicaUtilities.h"! within an \lstinline!Include! annotation (see \cref{annotations-for-external-libraries-and-include-files}); no \lstinline!IncludeDirectory! annotation is needed.
+
+\begin{example}
+The following usage patterns are common:
+\begin{itemize}
+\item
+  The \lstinline!Include! annotation may first \lstinline[language=C]!#include "ModelicaUtilities.h"!, and then contain the external function definition where the utility functions may be used.
+\item
+  Like above, but moving the annotation content above to, say, \filename{myExtFun.c}, and then just do \lstinline[language=C]!#include "myExtFun.c"! in the annotation.
+\item
+  After \lstinline[language=C]!#include "ModelicaUtilities.h"!, the \lstinline!Include! annotation can also define a wrapper around a function in a linked library that does the real job, where the wrapper forwards the arguments as well as passes function pointers for a selection of functions from \filename{ModelicaUtilities.h}.
+\end{itemize}
+\end{example}
+
+\begin{nonnormative}
+There is no standardized way to build external libraries using the functions in \filename{ModelicaUtilities.h}.
+For example, there is no standardized way of obtaining a directory where the tool shall provide the \filename{ModelicaUtilities.h}.
+There is also no standardized library to link with to get a tool-specific implementation of the functions.
```

Comment:
Based on https://github.com/modelica/ModelicaStandardLibrary/pull/4487
```suggestion
External libraries can built with the \filename{ModelicaUtilities.h} included within the Modelica Standard Library.
```
Since the resulting simulation model including the library will be linked with the tool-specific library by the tool I'm not sure how to mention that part.